### DISCUSSION TITLE
1.8.6

### DISCUSSION THREAD

**Discussion by saejin-moon** [thelegendski/ski.js#17] [Announcements]
`print.js` changed drastically.
all transformations got a makeover.
`Canvas` was removed.
variable names were changed so you can actually use `data` w/o screwin' with ski.js lol.
mouse movement was fixed.
credit was added.

currently workin' on responsive canvas capabilities.
please leave any feedback in the discussion for this release or in a new issue, :D!

<hr /><em>This discussion was created from the release <a href='https://github.com/thelegendski/ski.js/releases/tag/v1.8.6'>1.8.6</a>.</em>

**Comment by 99TheDark** [comment]
What do you mean by 'responsive canvas capabilities'? Like resizing and such? Swapping the canvas?

**Reply by 99TheDark** [reply]
On a separate topic it's nice having the differences between each version with the auto compare

**Reply by saejin-moon** [reply]
resizin'. that's still a wip mostly 'cuz i haven't touched it.
def love auto compare. been keepin' up with the changes in Dark.js files to see what i can use over here, :p.

**Reply by 99TheDark** [reply]
Lol, I'm wondering if my 'oh crap forgot that' updates are getting spammy, I made like 5 updates in 20 minutes one day because I messed up so bad

**Reply by 99TheDark** [reply]
My resizing works fine if you want to look at that. It's annoying because you have to save and copy EVERY SINGLE SETTING in the context after resizing a canvas. Argh!

**Reply by saejin-moon** [reply]
dang, i hadn't thought 'bout that. it makes zero sense why it would do that ngl.

what 'bout creatin' a new offscreen canvas at the new width an' height draw the ol' canvas on top an' apply all the ol' settin's then set the canvas displayed by the canvas tag to the offscreen canvas? wait, that wouldn't work.

**Reply by 99TheDark** [reply]
Here's my code: 
```
size = function(w = innerWidth, h = innerHeight) {
    if(typeof w == "number" && typeof h == "number" && w > 0 && h > 0) {
        // Because for some reason changing width & height reset all parameters >:(
        // It took me ~8 hours to figure this out. D:<
        let old = d.copy(d.ctx);

        [d.width, d.height] = [d.canvas.width, d.canvas.height] = [w, h];

        for(const key in d.ctx) {
            const value = old[key];
            if(typeof value !== "function" && !Dark.styleIgnore.includes(key)) {
                d.ctx[key] = value;
            }
        }
        Dark.globallyUpdateVariables(d);
    }
};
```

Btw, `d` = `this` and `copy` is a function that sets all the key value pairs by using a `for in` loop.

**Reply by saejin-moon** [reply]
only question is whether this works when the user makes the window smaller than larger. it looks like it should, but just checkin'.

**Reply by 99TheDark** [reply]
I mean it shouldn't have any difference. Just remember mine is a resizable canvas element and doesn't have auto resize.

Yay, I get to be a contributor by doing nothing! (newest release)

**Reply by saejin-moon** [reply]
alr then, it'll be headed in `1.9.0`, :D!

give credit where it is due, :).

**Reply by 99TheDark** [reply]
Aha, you contributed nothing! I didn't even realize you could add contributors to releases. I thought that's what CREDITS.md was for. Good luck with that! I just got kernels to work, kind annoying how limited it is. I can't say make an array of x size, just something like 14.

**Reply by 99TheDark** [reply]
I suppose this conversation is heading more towards ski & dark chat

**Reply by saejin-moon** [reply]
movin' it back over to the other thread, :p.
